libminifi/include/utils/StoppableThread.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <mutex>
+#include <thread>
+#include <condition_variable>
+#include <atomic>
+#include <functional>
+
+namespace org::apache::nifi::minifi::utils {
+
+// mimics some aspects of std::jthread
+// unfortunately clang's jthread support is lacking
+// TODO(adebreceni): replace this with std::jthread
+class StoppableThread {
+ public:
+  explicit StoppableThread(std::function<void()> fn);
+
+  void stopWait() {
```

Comment:
Wouldn't the name `stop()` be more clear?

Reply:
comparing this with std::jthread::request_stop, that does not wait for the thread to terminate, I think it is important to emphasise that this waits

Reply:
I think it the name `stop` is already distinguished in its name as jthread only requests the stop and this one actually stops, but if you think it needs more emphasis maybe a `stopAndWait`, `stopWithWait`, or something similar would be better as I associate this name with a functionality like "stop the waiting" :smile:

Reply:
renamed